packages/serve/src/lib/middleware/auth/authCredentialsMiddleware.ts

```diff
@@ -15,8 +40,19 @@ export class AuthCredentialsMiddleware extends BaseAuthMiddleware {
     // return to stop the middleware, if disabled
     if (!this.enabled) return next();
 
-    // The /auth/token endpoint not need contains auth credentials
-    if (context.path === '/auth/token') return next();
+    // The endpoint not need contains auth credentials
+    const docPrefix =
+      this.projectOptions?.['redoc']?.url
+        .replace(/\/$/, '')
+        .replace(/^\//, '') || 'doc';
+    const pathsWithoutAuth = [
+      '/auth/token',
+      '/auth/available-types',
+      `/${docPrefix}`,
+      `/${docPrefix}/spec`,
+      `/${docPrefix}/redoc`,
+    ];
+    if (pathsWithoutAuth.includes(context.path)) return next();
```

Comment:
Seems the AuthSourceNormalizerMiddleware also has the same code, maybe we could refactor the part to a function and put it in utils.ts under the auth folder, how do you think?

Reply:
Agree, the code had been refactored in [1f05032](https://github.com/Canner/vulcan-sql/pull/160/commits/1f050324c8d77e7afb547d84d47be8e182174bf5)